File: passenger-service/src/main/java/com/software/modsen/passengerservice/repository/PassengerRatingRepository.java
package com.software.modsen.passengerservice.repository;

import com.software.modsen.passengerservice.entity.PassengerRating;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

import java.util.Optional;

@Repository
public interface PassengerRatingRepository extends JpaRepository<PassengerRating, Long> {
    Optional<PassengerRating> findPassengerRatingByPassengerId(Long passengerId);
}

File: passenger-service/src/main/kotlin/com/software/modsen/passengerservice/repository/PassengerRatingRepository.kt
package com.software.modsen.passengerservice.repository

import com.software.modsen.passengerservice.entity.PassengerRating
import jakarta.persistence.LockModeType
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.data.jpa.repository.Lock
import org.springframework.stereotype.Repository

@Repository
interface PassengerRatingRepository : JpaRepository<PassengerRating, Long> {
    @Lock(value = LockModeType.OPTIMISTIC)
    fun findPassengerRatingByPassengerId(passengerId: Long): PassengerRating?
}